src/OrchardCore.Modules/OrchardCore.Email.Azure/Services/AzureEmailProviderBase.cs

```diff
@@ -224,4 +226,33 @@ private EmailMessage FromMailMessage(MailMessage message, Dictionary<string, ILi
 
         return emailMessage;
     }
+
+    private static EmailAddress ConvertEmailAddressToAzureEmailAddress(string emailWithDisplayName)
+    {
+        var (displayName, emailAddress) = ParseEmailAddressWithDisplayName(emailWithDisplayName);
+
+        return new EmailAddress(emailAddress, displayName);
+    }
+
+    private static (string DisplayName, string EmailAddress) ParseEmailAddressWithDisplayName(string emailWithDisplayName)
+    {
+        var match = AzureEmailProviderBaseRegexes.ParseEmailAddressWithDisplayNameRegex().Match(emailWithDisplayName);
```

Comment:
Instead of manually parsing the addresses, consider using `MailboxAddress` from MimeKit, e.g. like in the `EmailAddressValidator`.

Reply:
I have, but I wouldn't add that as a dependency just for this.

Reply:
MailAddress.TryCreate FTW, thanks! It does everything we need (and more than what I have here).